File: Módulo 1/Proramação & Algoritimo/Loops/Calculadora de Média/main.lua
local media = 0.0;
local soma_notas = 0.0;
local situacao = "";

for (local i = 0; i < 4; i = i + 1) do
    print("Insira nota " + i + ": ");
    local nota = io.read("*n");

    if (nota >= 0 && nota <= 10) do
        soma_notas += nota;
    else do
        print("Insira uma nota válida (1-10)...");
        i -= 1;
    end
end

media = soma_notas / 4;
if (media >= 7) do
    situacao = "Aprovado";
else do
    situacao = "Reporvado";
end

print("\nMédia final: " .. media .. "\nSituação: " .. situacao);

File: Módulo 1/Proramação & Algoritimo/Loops/Calculadora de Média/Java/Main.java
import java.util.Scanner;
public class Main {
	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);		

		double soma_das_notas = 0.0;
		double media = 0.0;
		String situacao = "";
		
		for (int i = 0; i < 4; i++) {
			double nota = 0.0;
			System.out.print("\nInsira a nota " + (i + 1) + ": ");
			nota = scanner.nextDouble();
			
			if (nota <= 10 && nota > 0) {
				soma_das_notas += nota;
			}
			else {
				System.out.println("Erro. Insira um número válido...");
				i -= 1;
			}
		}

		media = soma_das_notas / 4;
		if (media >= 7) {
			situacao = "Aprovado";
		}
		else {
			situacao = "Reprovado";
		}

		// Output
		System.out.println("\nMédia final: " + media);
		System.out.println("\nSituação: " + situacao);

		scanner.close();
	}
}